### PR TITLE
[hail] let

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

File: hail/python/hail/expr/functions.py [hunk 1]
```diff
@@ -331,6 +331,41 @@ def bind(f: Callable, *exprs):
     return construct_expr(res_ir, lambda_result.dtype, indices, aggregations)
 
 
+def let(**exprs):
```
Comment (tpoterba) [inline hail/python/hail/expr/functions.py]: this should be `(*exprs)` right?

Comment (danking) [reply hail/python/hail/expr/functions.py]: fixed

**Comment by tpoterba** [comment]
I'm a bit worried about confusing people with two (almost) identical methods named differently. 

What do you think about something like `rbind` for right-bind? Then it's clear they're in the same family

**Comment by tpoterba** [comment]
happy to rename to `let` in 0.3

**Pull request merged**